src/Microsoft.ML.Data/Model/Onnx/IOnnxNode.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Model.Onnx
+{
+    /// <summary>
+    /// An abstraction for an ONNX node as created by
+    /// <see cref="IOnnxContext.CreateNode(string, IEnumerable{string}, IEnumerable{string}, string, string)"/>.
+    /// That method creates a with inputs and outputs, but this object can modify the node further
+    /// by adding attributes (in ONNX parlance, attributes are more or less constant parameterizations).
+    /// </summary>
+    public interface IOnnxNode
+    {
+        void AddAttribute(string argName, double value);
+        void AddAttribute(string argName, IEnumerable<double> value);
+        void AddAttribute(string argName, IEnumerable<float> value);
```

Comment:
>IEnumerable<float> value); [](start = 42, length = 26)

so everyone has a pair of T value and IEnumerable<T> method, but float has only IEnumerable<T>
I don't know is it wrong or not, but it just stands out.

Reply:
Yes. I think it's because `double` param will work for both `float` and `double`, but `IEnumerable<double>` will only work for `IEnumerable<double>`, not also `IEnumerable<float>`.